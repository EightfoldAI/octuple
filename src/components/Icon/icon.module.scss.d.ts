declare namespace IconModuleScssModule {
    export interface IIconModuleScss {
        body1: string;
        body2: string;
        body3: string;
        'button-spacer-1': string;
        'button-spacer-2': string;
        'button-spacer-3': string;
        button1: string;
        button2: string;
        button3: string;
        buttonSpacer1: string;
        buttonSpacer2: string;
        buttonSpacer3: string;
        caption: string;
        'chevron-icon': string;
        'chevron-spacer': string;
        chevronIcon: string;
        chevronSpacer: string;
        'custom-padding-vertical': string;
        customPaddingVertical: string;
        display1: string;
        display2: string;
        display3: string;
        display4: string;
        'divider-line': string;
        dividerLine: string;
        'flex-structure-horizontal': string;
        'flex-structure-vertical': string;
        flexStructureHorizontal: string;
        flexStructureVertical: string;
        'full-height': string;
        fullHeight: string;
        header1: string;
        header2: string;
        header3: string;
        header4: string;
        header5: string;
        header6: string;
        icon: string;
        'icon-1-material': string;
        'icon-2-material': string;
        'icon-3-material': string;
        'icon-4-material': string;
        'icon-material-l': string;
        'icon-material-m': string;
        'icon-material-s': string;
        'icon-material-xs': string;
        icon1Material: string;
        icon2Material: string;
        icon3Material: string;
        icon4Material: string;
        iconMaterialL: string;
        iconMaterialM: string;
        iconMaterialS: string;
        iconMaterialXs: string;
        iconWrapper: string;
        l: string;
        'logo-text': string;
        logoText: string;
        m: string;
        mappings: string;
        names: string;
        'octuple-menu': string;
        octupleMenu: string;
        'open-sans': string;
        openSans: string;
        'pre-display': string;
        preDisplay: string;
        s: string;
        semibold: string;
        sourceRoot: string;
        sources: string;
        sourcesContent: string;
        'space-m': string;
        spaceM: string;
        spacer: string;
        subtitle1: string;
        subtitle2: string;
        version: string;
        xl: string;
        xxl: string;
        xxxl: string;
    }
}

declare const IconModuleScssModule: IconModuleScssModule.IIconModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: IconModuleScssModule.IIconModuleScss;
};

export = IconModuleScssModule;
