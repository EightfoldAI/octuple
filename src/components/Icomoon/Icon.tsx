import React, { FC, useContext } from 'react';
import { mergeClasses } from '../../shared/utilities';
import { IcomoonProps, IcomoonSize } from './';
import IcomoonReact from 'icomoon-react';

import styles from './icon.module.scss';

const IconSetContext = React.createContext<object | null>(null);

export const IcomoonProvider = IconSetContext.Provider;

export const Icomoon: FC<IcomoonProps> = ({
    ariaHidden = false,
    classNames,
    color,
    iconName,
    id,
    role = 'presentation',
    size = IcomoonSize.Medium,
    'data-test-id': dataTestId,
}) => {
    const iconClassNames: string = mergeClasses([
        classNames,
        styles.iconWrapper,
    ]);

    return (
        <span
            data-test-id={dataTestId}
            aria-hidden={ariaHidden}
            className={iconClassNames}
            id={id}
            role={role}
        >
            <IconSetContext.Consumer>
                {(iconSet) => (
                    <IcomoonReact
                        iconSet={iconSet}
                        icon={iconName}
                        size={size}
                        color={color}
                    />
                )}
            </IconSetContext.Consumer>
        </span>
    );
};
