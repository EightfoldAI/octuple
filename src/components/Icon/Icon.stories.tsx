import React from 'react';
import { Icon, IconName, IconSize } from './index';

export default {
    title: 'Icon',
    component: Icon,
};

export const Icons = () => (
    <>
        <h2>Icons</h2>
        <p>X-Small</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.XSmall} />
        <p>Small</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.Small} />
        <p>Medium</p>
        <Icon path={IconName.mdiCardsHeart} />
        <p>Large</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.Large} />
    </>
);
