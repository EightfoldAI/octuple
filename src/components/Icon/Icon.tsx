'use client';

import React, { FC } from 'react';
import IcomoonReact from 'icomoon-react';

import { mergeClasses } from '../../shared/utilities';
import { Icon as MdiIcon } from '@mdi/react';
import { IconProps, IconSize } from './index';

import styles from './icon.module.scss';
import { useConfig } from '../ConfigProvider';

export const Icon: FC<IconProps> = ({
  ariaHidden = false,
  classNames,
  color,
  description,
  horizontal,
  id,
  path,
  role = 'presentation',
  rotate,
  size = IconSize.Medium,
  spin,
  style,
  title,
  vertical,
  'data-test-id': dataTestId,
  icomoonIconName,
}) => {
  const { icomoonIconSet } = useConfig();

  const iconClassNames: string = mergeClasses([classNames, styles.iconWrapper]);

  if (!icomoonIconName && !path) {
    return null;
  }

  const iconComponent = icomoonIconName ? (
    <IcomoonReact
      iconSet={icomoonIconSet}
      size={size}
      color={color}
      icon={icomoonIconName}
      role="presentation"
    />
  ) : (
    <MdiIcon
      color={color}
      description={description}
      horizontal={horizontal}
      path={path}
      rotate={rotate}
      size={size}
      title={title}
      vertical={vertical}
      spin={spin}
    />
  );

  return (
    <span
      data-test-id={dataTestId}
      aria-hidden={ariaHidden}
      className={iconClassNames}
      id={id}
      role={role}
      style={style ? style : null}
    >
      {iconComponent}
    </span>
  );
};

Icon.displayName = 'Icon';
