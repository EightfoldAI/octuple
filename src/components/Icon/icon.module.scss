.iconWrapper {
    display: inline-block;
    margin: 0;
    padding: 0;

    svg {
        vertical-align: middle;
    }
}