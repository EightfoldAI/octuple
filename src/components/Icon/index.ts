export * from './Icon';
export * from './Icon.types';
export * from './mdi';
