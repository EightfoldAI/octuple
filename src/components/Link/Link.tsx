'use client';

import React, { Ref, FC, useContext } from 'react';
import DisabledContext, { Disabled } from '../ConfigProvider/DisabledContext';
import { OcThemeName } from '../ConfigProvider';
import ThemeContext, {
  ThemeContextProvider,
} from '../ConfigProvider/ThemeContext';
import { LinkProps } from './Link.types';
import { mergeClasses } from '../../shared/utilities';

import styles from './link.module.scss';
import themedComponentStyles from './link.theme.module.scss';

export const Link: FC<LinkProps> = React.forwardRef(
  (
    {
      href,
      classNames,
      children,
      configContextProps = {
        noDisabledContext: false,
        noThemeContext: false,
      },
      disabled = false,
      fullWidth = true,
      onClick,
      role = 'link',
      target = '_self',
      underline,
      variant = 'default',
      style,
      theme,
      themeContainerId,
      'data-test-id': dataTestId,
      ...rest
    },
    ref: Ref<HTMLAnchorElement>
  ) => {
    const contextuallyDisabled: Disabled = useContext(DisabledContext);
    const mergedDisabled: boolean = configContextProps.noDisabledContext
      ? disabled
      : contextuallyDisabled || disabled;

    const contextualTheme: OcThemeName = useContext(ThemeContext);
    const mergedTheme: OcThemeName = configContextProps.noThemeContext
      ? theme
      : contextualTheme || theme;

    const linkClassNames: string = mergeClasses([
      styles.linkStyle,
      { [styles.fullWidth]: !!fullWidth },
      { [styles.neutral]: variant === 'neutral' },
      { [styles.primary]: variant === 'primary' },
      { [styles.secondary]: variant === 'secondary' },
      { [styles.disruptive]: variant === 'disruptive' },
      { [styles.underline]: !!underline },
      { [themedComponentStyles.theme]: mergedTheme },
      { [styles.disabled]: mergedDisabled },
      classNames,
    ]);

    const handleOnClick = (
      event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
    ): void => {
      if (mergedDisabled) {
        event.preventDefault();
        return;
      }
      onClick?.(event);
    };

    return (
      <ThemeContextProvider
        componentClassName={themedComponentStyles.theme}
        containerId={themeContainerId}
        theme={mergedTheme}
      >
        <a
          {...rest}
          ref={ref}
          role={role}
          aria-disabled={mergedDisabled}
          className={linkClassNames}
          href={href}
          onClick={handleOnClick}
          style={style}
          target={target}
          data-test-id={dataTestId}
        >
          {children}
        </a>
      </ThemeContextProvider>
    );
  }
);
