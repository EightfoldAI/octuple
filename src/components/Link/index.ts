export * from './Navlink.types';
export * from './Navlink';
