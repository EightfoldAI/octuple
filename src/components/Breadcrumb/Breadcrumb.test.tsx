import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Breadcrumb } from './';
import { IconName } from '../Icon';
import { Link } from '../Link';
import { render, screen } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

class ResizeObserver {
  observe() {
    // do nothing
  }
  unobserve() {
    // do nothing
  }
  disconnect() {
    // do nothing
  }
}

window.ResizeObserver = ResizeObserver;

const myLinks = [
  {
    title: 'Home',
    url: '#',
  },
  {
    title: 'Page 1',
    url: '#',
  },
  {
    title: 'Page 2',
    url: '#',
  },
  {
    title: 'Page 3',
    url: '#',
  },
  {
    title: 'Page 4',
    url: '#',
  },
  {
    title: 'Page 5',
    url: '#',
  },
  {
    title: 'Page 6',
    url: '#',
  },
];

const breadcrumbArgs = {
  ariaLabel: 'Breadcrumbs',
  classNames: 'my-breadcrumb-root-class',
  'data-testid': 'my-breadcrumb-testid',
  divider: {
    path: IconName.mdiChevronRight,
    ariaHidden: true,
    classNames: 'my-breadcrumb-icon',
    id: 'myBreadcrumbIcon',
    role: 'presentation',
    rotate: 0,
    spin: false,
    vertical: false,
    'data-testid': 'myBreadcrumbIconTestId',
  },
  id: 'myBreadcrumb',
  links: myLinks,
  linkClassNames: 'my-breadcrumb-links-class',
  maxDisplayedLinks: 3,
  overflowAriaLabel: 'More links',
  style: {},
};

describe('Breadcrumb', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container } = render(<Breadcrumb {...breadcrumbArgs} />);
    expect(() => container).not.toThrowError();
    expect(container.querySelector('.breadcrumb')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('should accept undefined links', () => {
    const { asFragment } = render(
      <Breadcrumb {...breadcrumbArgs} links={undefined!} />
    );
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  test('should support custom attribute', () => {
    const { asFragment } = render(
      (
        <Breadcrumb
          {...breadcrumbArgs}
          links={[
            {
              title: 'xxx',
              // @ts-ignore
              'data-custom': 'custom-item',
            },
            {
              title: 'yyy',
            },
          ]}
          data-custom="custom"
        />
      ) as React.ReactElement<any, string | React.JSXElementConstructor<any>>
    );
    expect(asFragment().firstChild).toMatchSnapshot();
  });

  test('Crumb should support string `0`', () => {
    const { container } = render(
      <Breadcrumb
        {...breadcrumbArgs}
        links={[
          {
            title: '0',
          },
        ]}
      />
    );
    expect(container.querySelectorAll('.breadcrumb-link')[0].textContent).toBe(
      '0'
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Crumb should use `onClick`', async () => {
    const onClick = jest.fn();
    const { container } = render(
      <Breadcrumb {...breadcrumbArgs} links={[{ title: 'test', onClick }]} />
    );
    const crumb = await screen.findByText('test');
    crumb.click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('Crumb should use `classNames`', async () => {
    const { container } = render(
      <Breadcrumb
        {...breadcrumbArgs}
        links={[{ title: 'test', classNames: 'my-breadcrumb-link-class' }]}
      />
    );
    const crumb = await screen.findByText('test');
    expect(crumb.classList.contains('my-breadcrumb-links-class')).toBe(true);
    expect(crumb.classList.contains('my-breadcrumb-link-class')).toBe(true);
  });

  test('Crumb should use custom links', async () => {
    const { container } = render(
      <Breadcrumb
        {...breadcrumbArgs}
        links={[
          {
            title: 'Home',
            children: (
              <Link variant="secondary" href="#">
                Custom home
              </Link>
            ),
          },
          {
            title: 'Page 1',
            children: (
              <Link variant="disruptive" href="#">
                Custom Page 1
              </Link>
            ),
            tooltipprops: {
              content: 'A tooltip',
            },
          },
          {
            ariaCurrent: true,
            readonly: true,
            title: 'Page 2',
            url: '#',
          },
        ]}
      />
    );
    expect(container.querySelector('.breadcrumb')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
