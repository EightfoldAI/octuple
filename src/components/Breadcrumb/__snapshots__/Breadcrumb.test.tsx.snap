// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Crumb should support string \`0\` and number \`0\` 1`] = `
<div
  aria-label="Breadcrumbs"
  class="breadcrumb my-breadcrumb-root-class"
  data-testid="my-breadcrumb-testid"
  id="myBreadcrumb"
  role="navigation"
>
  <ol
    class="breadcrumb-list"
  >
    <li>
      <div
        class="stack horizontal undefined gap-xxxs"
      >
        <div
          class="reference-wrapper disabled"
          id="tooltip-"
        >
          <a
            aria-disabled="false"
            class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
            data-reference-id="tooltip--reference"
            id="tooltip--reference"
            role="link"
            target="_self"
            title="0"
          >
            0
          </a>
        </div>
        <span
          aria-hidden="false"
          class="icon-wrapper"
          role="presentation"
        >
          <svg
            role="presentation"
            style="width: 16px; height: 16px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 21L14.9 3H17L9.1 21H7Z"
              style="fill: currentColor;"
            />
          </svg>
        </span>
      </div>
    </li>
    <li>
      <div
        class="stack horizontal undefined gap-xxxs"
      >
        <div
          class="reference-wrapper disabled"
          id="tooltip-"
        >
          <a
            aria-disabled="false"
            class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
            data-reference-id="tooltip--reference"
            id="tooltip--reference"
            role="link"
            target="_self"
            title="0"
          >
            0
          </a>
        </div>
      </div>
    </li>
  </ol>
</div>
`;

exports[`Breadcrumb Crumb should use custom links 1`] = `
<div>
  <div
    aria-label="Breadcrumbs"
    class="breadcrumb my-breadcrumb-root-class"
    data-testid="my-breadcrumb-testid"
    id="myBreadcrumb"
    role="navigation"
  >
    <ol
      class="breadcrumb-list"
    >
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width secondary"
              data-reference-id="tooltip--reference"
              href="#"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="undefined"
            >
              Custom home
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width disruptive"
              data-reference-id="tooltip--reference"
              href="#"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="undefined"
            >
              Custom Page 1
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <span
              aria-current="location"
              class="tooltip-reference breadcrumb-link my-breadcrumb-links-class breadcrumb-link-read-only"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
            >
              Page 2
            </span>
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Breadcrumb Renders without crashing 1`] = `
<div>
  <div
    aria-label="Breadcrumbs"
    class="breadcrumb my-breadcrumb-root-class"
    data-testid="my-breadcrumb-testid"
    id="myBreadcrumb"
    role="navigation"
  >
    <ol
      class="breadcrumb-list"
    >
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Home"
              url="#"
            >
              Home
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 1"
              url="#"
            >
              Page 1
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 2"
              url="#"
            >
              Page 2
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 3"
              url="#"
            >
              Page 3
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 4"
              url="#"
            >
              Page 4
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 5"
              url="#"
            >
              Page 5
            </a>
          </div>
          <span
            aria-hidden="false"
            class="icon-wrapper"
            role="presentation"
          >
            <svg
              role="presentation"
              style="width: 16px; height: 16px;"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 21L14.9 3H17L9.1 21H7Z"
                style="fill: currentColor;"
              />
            </svg>
          </span>
        </div>
      </li>
      <li>
        <div
          class="stack horizontal undefined gap-xxxs"
        >
          <div
            class="reference-wrapper disabled"
            id="tooltip-"
          >
            <a
              aria-disabled="false"
              class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
              data-reference-id="tooltip--reference"
              id="tooltip--reference"
              role="link"
              target="_self"
              title="Page 6"
              url="#"
            >
              Page 6
            </a>
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Breadcrumb should accept undefined links 1`] = `
<div
  aria-label="Breadcrumbs"
  class="breadcrumb my-breadcrumb-root-class"
  data-testid="my-breadcrumb-testid"
  id="myBreadcrumb"
  role="navigation"
>
  <ol
    class="breadcrumb-list"
  />
</div>
`;

exports[`Breadcrumb should support custom attribute 1`] = `
<div
  aria-label="Breadcrumbs"
  class="breadcrumb my-breadcrumb-root-class"
  data-custom="custom"
  data-testid="my-breadcrumb-testid"
  id="myBreadcrumb"
  role="navigation"
>
  <ol
    class="breadcrumb-list"
  >
    <li>
      <div
        class="stack horizontal undefined gap-xxxs"
      >
        <div
          class="reference-wrapper disabled"
          id="tooltip-"
        >
          <a
            aria-disabled="false"
            class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
            data-custom="custom-item"
            data-reference-id="tooltip--reference"
            id="tooltip--reference"
            role="link"
            target="_self"
            title="xxx"
          >
            xxx
          </a>
        </div>
        <span
          aria-hidden="false"
          class="icon-wrapper"
          role="presentation"
        >
          <svg
            role="presentation"
            style="width: 16px; height: 16px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 21L14.9 3H17L9.1 21H7Z"
              style="fill: currentColor;"
            />
          </svg>
        </span>
      </div>
    </li>
    <li>
      <div
        class="stack horizontal undefined gap-xxxs"
      >
        <div
          class="reference-wrapper disabled"
          id="tooltip-"
        >
          <a
            aria-disabled="false"
            class="link-style full-width primary tooltip-reference breadcrumb-link my-breadcrumb-links-class"
            data-reference-id="tooltip--reference"
            id="tooltip--reference"
            role="link"
            target="_self"
            title="yyy"
          >
            yyy
          </a>
        </div>
      </div>
    </li>
  </ol>
</div>
`;
